src/U1Manager.cpp

```
@@ -0,0 +1,397 @@
+    // the firmware's own OpenSpool parser uses for OpenSpool tags.
+    if (p.min_bed_temp > 0)      info.bed_temp = p.min_bed_temp;
+    else if (p.max_bed_temp > 0) info.bed_temp = p.max_bed_temp;
+
+    packCardUid(p.spool_id, info.card_uid, sizeof(info.card_uid), info.card_uid_len);
+    applyMaterialDefaults(info);
+    return info;
+}
+
+// Build U1 wire info from a Spoolman sync result (generic UID tag path) plus
+// the current NFC state for fallback fields the sync may not have populated.
+U1FilamentInfo buildFromSpoolmanSync(const SpoolmanSyncedPayload& s,
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)